### ISSUE TITLE
[libc++] packaged_task(std::allocator_arg_t, const Allocator&, F&&) should be gone in C++17 and later

### ISSUE THREAD

**Issue by Quuxplusone** [llvm/llvm-project#53912]
https://en.cppreference.com/w/cpp/thread/packaged_task/packaged_task
libc++ still accepts this `packaged_task` constructor in C++17 mode, but it shouldn't: https://godbolt.org/z/Mn6M5zo8a
```
#include <future>

struct A { A(int); };
using PA = std::packaged_task<A(int)>;
using PI = std::packaged_task<int(int)>;

static_assert(!std::is_constructible<PA, std::allocator_arg_t, std::allocator<A>, const PA&>::value, "");
static_assert( std::is_constructible<PA, std::allocator_arg_t, std::allocator<A>, const PI&>::value, ""); // should be false in C++17
static_assert( std::is_constructible<PA, std::allocator_arg_t, std::allocator<A>, int(*)(int)>::value, ""); // should be false in C++17
```
This was discovered incidentally while working on https://reviews.llvm.org/D119958

Whoever looks into this should find out what paper removed this ctor, and whether it removed (or added!) anything else too. We might be missing several related things at once.

**Comment by jwakely** [comment]
N.B. I'm proposing to restore this constructor, so you might want to wait ...

**Comment by frederick-vs-ja** [comment]
This should have been fixed by #107960. But as mentioned by @jwakely, the changes need to be reverted if [LWG3003](https://cplusplus.github.io/LWG/issue3003) gets accepted.

**Comment by ldionne** [comment]
Closing since this was technically resolved by removing the constructors as mentioned by @frederick-vs-ja. Let's just handle this via our usual LWG issue process. We might end up flip-flopping users a little bit, but I really don't think these constructors are seeing a lot of usage so I don't think the impact is going to be noticeable.
